library/cartridge_edit_topology.py

```diff
@@ -407,51 +454,80 @@ def get_servers_params(instances, cluster_instances, allow_missed_instances):
+    if changed_advertise_uris and not ignore_checks.get(ADVERTISE_URIS_CHANGE_CHECK_NAME):
+        msg = 'Advertise uris of some instances changed: %s' % ', '.join(changed_advertise_uris)
+        errors.append(msg)
 
-    return topology_params, None
+    return errors
 
 
-def get_replicasets_failover_priority_and_instances_params(
-    replicasets, cluster_replicasets, instances, cluster_instances, allow_missed_instances
-):
-    topology_params = {}
+def check_replicasets_for_dangerous_changes(all_replicasets, replicasets, cluster_replicasets, ignore_checks):
+    errors = []
+    extra_cluster_replicasets = []
+    renamed_replicasets = []
 
-    replicasets_params, err = get_replicasets_params_for_changing_failover_priority(
-        replicasets, cluster_replicasets, cluster_instances, allow_missed_instances
-    )
-    if err is not None:
-        return None, err
+    for cluster_replicaset_name, cluster_replicaset_params in cluster_replicasets.items():
+        if cluster_replicaset_name not in all_replicasets:
+            renamed = False
 
-    if replicasets_params:
-        topology_params['replicasets'] = replicasets_params
+            for replicaset_name, replicaset_params in all_replicasets.items():
+                if set(replicaset_params.get('instances', [])) == set(cluster_replicaset_params.get('instances', [])):
+                    renamed = True
+                    if replicaset_name in replicasets:
+                        renamed_replicasets.append('%s -> %s' % (replicaset_name, cluster_replicaset_name))
 
-    # manage instances that were joined on previous call
-    servers_params, err = get_servers_params(instances, cluster_instances, allow_missed_instances)
-    if err is not None:
-        return None, err
+            if not renamed:
+                extra_cluster_replicasets.append(cluster_replicaset_name)
 
-    if servers_params:
-        topology_params['servers'] = servers_params
+    if extra_cluster_replicasets and not ignore_checks.get(EXTRA_CLUSTER_REPLICASETS_CHECK_NAME):
+        msg = 'Some replicasets from cluster are missing in inventory: %s' % ', '.join(extra_cluster_replicasets)
+        errors.append(msg)
 
-    return topology_params, None
+    if renamed_replicasets and not ignore_checks.get(RENAMED_REPLICASETS_CHECK_NAME):
+        msg = 'Looks like that some replicasets should be renamed in inventory: %s' % ', '.join(renamed_replicasets)
+        errors.append(msg)
+
+    return errors
+
+
+def check_for_dangerous_changes(
+    all_instances, instances, cluster_instances,
+    all_replicasets, replicasets, cluster_replicasets,
+    ignore_checks,
+):
+    errors = []
+    errors += check_instances_for_dangerous_changes(all_instances, instances, cluster_instances, ignore_checks)
+    errors += check_replicasets_for_dangerous_changes(all_replicasets, replicasets, cluster_replicasets, ignore_checks)
+    if errors:
+        return '. '.join(errors)
```

Comment:
Maybe `"Found dangerous changes: %s. To ignore some of them use these flags..."`